docs/developing/integrate/neon_transfer_sdk.md

```diff
@@ -4,30 +4,54 @@ proofedDate: na
+comment: todo this requires overhaul 
 ---
 
-**NOTE**
+This page provides you with the information required to pass tokens between Solana and Neon EVM programmatically. 
 
-The package is tested on [NeonPass](https://neonpass.live/) codebase, which use react hooks arch.
-You can use clean core, but all important requirements you need to provide as params and see how it goes by yourself.
+## TL;DR
+
+- The tool is available as an npm package
+- The package provides the same function as the [NeonPass UI](https://devnet.neonpass.live)
+
+## Introduction
+
+The following npm package provides a token pass tool. This same package provides the functionality to the NeonPass UI tool. It is built to be compliant with the React hooks architecture, but you are welcome to use clean core functions.
+
+You need to provide the addresses for a Solana-compatible wallet such as the Phantom wallet and an EVM-compatible wallet such as the MetaMask wallet. Both of these addresses are passed as required properties. The codule checks whether both wallet objects - window.solana and window.ehtereum - really exist. If they don't, the module will throw an error for you. 
 
----
 
-Module was used by the react wrapper `import {useNeonTransfer} from "neon-portal/src/react"`
 For clean working configuration example we have to rebuild connect status buttons, their transfer callbacks and error handling.
 
-## Installation and setup
+## Step 1: installation and setup
 
 Firstly, install the package:
 
 ```sh
 npm install neon-portal --save
 ```
 
-### For native
+## Step 2: understand the parameters
+
+The following table provides details of the required and optional parameters. 
 
-After installing you need to pass some required properties, when calling new instance, for properly working. Neon wallet interface based on Metamask. Module checks, is wallet objects - window.solana and window.ehtereum - really exists. If not - module will throw an error for you.  So, besides Solana Phantom wallet you need to connect Metamask wallet and get both user addresses, which you need to pass as properties.
+
+|Property name |   Type|Description| is required |
+|:-------------|:------:|:------------------------------------------------------------------------------:|:------------|
+|solanaWalletAddress |  String  |Address for Solana-compatible wallet  |        true |
+|neonWalletAddress |  String  | Address of existing EVM-compatible wallet associated with a Neon Account | true |
+|web3 | Object| Connection module of Ethereum Web3.js framework. Connects the App to the Neon EVM network.| true |
+|connection |  Object  | Connection module of Solana-Web3.js framework. You may connect to your own node if you have one. Else you will be provided with the default RPC.|        true |
+|onBeforeCreateInstructions | function |Function executed at the start of transfer intitation. It is used in both transfer functions (`createNeonTransfer` and `createSolanaTransfer`).  | false |
```

Comment:
@oable Can you assist with this optional function pls -- it is not clear what its purpose is yet